pkg/network/driver_interface.go

```diff
@@ -415,6 +434,38 @@ func (dh *DriverHandle) setFilters(filters []C.struct__filterDefinition) error {
 	}
 	return nil
 }
+func minUint64(a, b uint64) uint64 {
+	if a > b {
+		return b
+	}
+	return a
+}
+
+// setParams passes any configuration values from the config file down
+// to the driver.
+func (di *DriverInterface) setFlowParams() error {
+	// set up the maximum flows
+
+	// temporary setup.  Will set the maximum flows to the sum of the configured
+	// max_tracked_connections and max_closed_connections_buffered, setting a
+	// (hard_coded) maximum.  This will be updated to actually honor the separate
+	// config values when the driver is updated to track them separately.
+
+	// this makes it so that the config can clamp down, but can never make it
+	// larger than the coded defaults above.
+	maxFlows := minUint64((defaultMaxOpenFlows + defaultMaxClosedFlows), (di.maxOpenFlows + di.maxClosedFlows))
+	log.Debugf("Setting max flows in driver to %v", maxFlows)
+	err := windows.DeviceIoControl(di.driverFlowHandle.handle,
+		C.DDNPMDRIVER_IOCTL_SET_MAX_FLOWS,
+		(*byte)(unsafe.Pointer(&maxFlows)),
+		uint32(unsafe.Sizeof(maxFlows)),
+		nil,
+		uint32(0), nil, nil)
+	if err != nil {
+		log.Warnf("Failed to set max number of flows to %v %v", maxFlows, err)
```

Comment:
Why?
If you don't log the error, and just return the error, when you get a flare from the customer all you have is the top-of-stack processing that indicates that an error occurred, and you don't have the indication of where the error actually occurred (this happens _all the time_).  If you log the error at the point of the error, then there's no guessing.

Reply:
The main reason is to prevent double-processing of the same error. If you log and return the error, and the parent function also logs and returns, and so on, you now have 2+ logs for the same error, which is noisy and not helpful. 

In go, errors are simply strings and don't have stack trace information. It is idiomatic to have unique content in your error messages to the amount of location disambiguation you need/want, rather than relying upon a log framework including the location.

This is why errors are usually wrapped (i.e. `return fmt.Errorf("error doing something: %w", err)`. The final error message ends up being `a: b: c: d`. This allows you to get a rough stack without paying the high cost of stack unwinding.